sim/midas/src/main/cc/bridges/termination.cc

```diff
@@ -0,0 +1,64 @@
+#ifdef TERMINATIONBRIDGEMODULE_struct_guard
+
+#include "termination.h"
+#include <iostream>
+
+termination_t::termination_t(
+	simif_t* sim, 
+  std::vector<std::string> &args,
+  TERMINATIONBRIDGEMODULE_struct * mmio_addrs,
+	unsigned int num_messages,
+  unsigned int* isErr,
+	const char* const* msgs): 
+	bridge_driver_t(sim), 
+	mmio_addrs(mmio_addrs), 
+	isErr(isErr),
+	msgs(msgs)
+{
+  this->mmio_addrs = mmio_addrs;
+  std::string tick_rate_arg = std::string("+tick-rate=");
+  std::string termination_rounds_arg = std::string("+termination-rounds=");
+  for (auto &arg: args) {
+		if (arg.find(tick_rate_arg) == 0) {
+  	  char *str = const_cast<char*>(arg.c_str()) + tick_rate_arg.length();
+  	  uint64_t tick_period = atol(str);
+  	  this->tick_rate = tick_period;
+		}
+  	if (arg.find(termination_rounds_arg) == 0) {
+  	  char *str = const_cast<char*>(arg.c_str()) + termination_rounds_arg.length();
+  	  uint64_t rounds = atol(str);
+  	  this->termination_rounds = rounds;
+		}
+	}
+}
+
+termination_t::~termination_t() {
+	free(this->mmio_addrs);
+}
+
+void termination_t::tick() {  //reads the MMIOs every tick
+  if(counter == this->tick_rate) {
+		if (read(this->mmio_addrs->out_status)) {
+		  msg_id = read(this->mmio_addrs->out_errCode);
+			if(this->isErr[msg_id]==0) {
+			  fail = 0;
+				std::cerr << "*PASS*" << std::endl;
+			} else {
+			  fail = 1;
+				std::cerr << "*FAIL*" << std::endl;
+			}
+			std::cerr << "Test completed with message";
```

Comment:
I think this would be better as: termination condition set on Cycle (return cycle). 

You'll need to add a cycle counter in your bridge, and then look at printf / assertion bridges to see how to get clock domain information.

Reply:
Done.